pkg/healthcheck/healthcheck.go

```diff
@@ -932,3 +960,39 @@ func validateDataPlanePodReporting(pods []*pb.Pod) error {
 
 	return nil
 }
+
+func checkUnschedulablePods(pods []corev1.Pod) error {
+	var podName []string
+	for _, pod := range pods {
+		for _, condition := range pod.Status.Conditions {
+			if condition.Reason == "Unschedulable" {
+				podName = append(podName, pod.Name)
+			}
+		}
+	}
+
+	if len(podName) != 0 {
+		pods := strings.Join(podName, ",")
+		return fmt.Errorf("No node available for the pods %s", pods)
+	}
+
+	return nil
+}
+
+func checkPodSecurityPolicies(rst []v1beta1.ReplicaSet) error {
```

Comment:
How about instead of explicitly single out psp error, we only have the `checkUnschedulablePods()` above, where we check for the pod's phase. If the phase isn't one of `succeeded`, `completed`, `running` or `pending`, then we output the status, conditions and reasons?

Reply:
For the reason stated in my comment below, we need special handling for PSP blockages that looks into Replicasets.

Reply:
.. but I agree `checkUnschedulablePods` could be more generic and return the actual error string, not a hard-coded one.

Reply:
yes we can have one more check as mentioned in the above comment, just to keep things organised